lib/xray_creator.py

```diff
@@ -42,12 +42,8 @@ def _initialize_books(self):
 
         self._books = []
         for book_id in self._book_ids:
-            book = Book(self._db, book_id, self._gConnection, self._aConnection, formats=self._formats,
-                send_to_device=self._send_to_device, create_xray=self._create_xray, proxy=self._proxy,
-                https_address=self._https_address, https_port=self._https_port)
-            self._books.append(book)
-            self._gConnection = book._gConnection
-            self._aConnection = book._aConnection
+            self._books.append(Book(self._db, book_id, self._gConnection, self._aConnection, formats=self._formats,
+                send_to_device=self._send_to_device, create_xray=self._create_xray))
         
         self._total_not_failing = 0
         book_lookup = {}
```

Comment:
I think there is still more cleanup that could be done.  Book.__init__ is passed the two connections and stores them, yet book.create_xray_event is passing through gConnection, and returning the updated value (which now will never change).  So can remove that return from create_xray_event, and it already has the connection so doesn't need to pass it through.

Same comments for send_xrays_event.

Generally though this looks so much nicer :)